File: Android/tuicallkit-kt/src/main/java/com/tencent/qcloud/tuikit/tuicallkit/state/MediaState.kt
package com.tencent.qcloud.tuikit.tuicallkit.state

import com.tencent.cloud.tuikit.engine.common.TUICommonDefine
import com.tencent.cloud.tuikit.engine.common.TUICommonDefine.AudioPlaybackDevice
import com.trtc.tuikit.common.livedata.LiveData

class MediaState {
    var hasMicrophonePermission: LiveData<Boolean> = LiveData(false)
    var isMicrophoneMuted: LiveData<Boolean> = LiveData(false)
    var audioPlayoutDevice: LiveData<AudioPlaybackDevice> = LiveData(AudioPlaybackDevice.Speakerphone)

    var hasCameraPermission: LiveData<Boolean> = LiveData(false)
    var isCameraOpened: LiveData<Boolean> = LiveData(false)
    var isFrontCamera: LiveData<TUICommonDefine.Camera> = LiveData(TUICommonDefine.Camera.Front)

    fun reset() {
        hasMicrophonePermission.set(false)
        isMicrophoneMuted.set(false)
        audioPlayoutDevice.set(AudioPlaybackDevice.Speakerphone)

        hasCameraPermission.set(false)
        isCameraOpened.set(false)
        isFrontCamera.set(TUICommonDefine.Camera.Front)
    }
}

File: iOS/TUICallKit_Swift/State/MediaState.swift
//
//  MediaState.swift
//  Pods
//
//  Created by vincepzhang on 2025/2/6.
//

import RTCCommon
import RTCRoomEngine

enum AudioPlaybackDevice: UInt {
    case speakerphone = 0
    case earpiece = 1
    case bluetooth = 2
}

class MediaState: NSObject {
    
    let isCameraOpened: Observable<Bool> = Observable(false)
    
    let isFrontCamera: Observable<Bool> = Observable(true)

    let isMicrophoneMuted: Observable<Bool> = Observable(false)
    
    let audioPlayoutDevice: Observable<AudioPlaybackDevice> = Observable(AudioPlaybackDevice.earpiece)
    
    func reset() {
        isCameraOpened.value = false
        isFrontCamera.value = true
        isMicrophoneMuted.value = false
        audioPlayoutDevice.value = .earpiece
    }
}
